src/cpp/database/database_queue.cpp

```diff
@@ -0,0 +1,645 @@
+                        item.second->entity_count());
+                database_->insert(domain, entity, sample);
+            }
+            catch (const eprosima::statistics_backend::Exception& e)
+            {
+                //TODO: maybe some logging would be nice
+            }
+        }
+        break;
+        case StatisticsEventKind::EDP_PACKETS:
+        {
+            EdpCountSample sample;
+            EntityId domain;
+            EntityId entity;
+            queue_item_type item = front();
+            sample.src_ts = item.first;
+            try
+            {
+                process_sample_type(domain, entity, EntityKind::PARTICIPANT, static_cast<EntityCountSample&>(sample),
+                        item.second->entity_count());
+                database_->insert(domain, entity, sample);
+            }
+            catch (const eprosima::statistics_backend::Exception& e)
+            {
+                //TODO: maybe some logging would be nice
+            }
+        }
+        break;
+        case StatisticsEventKind::DISCOVERED_ENTITY:
+        {
+            DiscoveryTimeSample sample;
+            EntityId domain;
+            EntityId entity;
+            queue_item_type item = front();
+            sample.src_ts = item.first;
+            try
+            {
+                process_sample_type(domain, entity, EntityKind::PARTICIPANT, sample, item.second->discovery_time());
+                database_->insert(domain, entity, sample);
+            }
+            catch (const eprosima::statistics_backend::Exception& e)
+            {
+                //TODO: maybe some logging would be nice
+            }
+        }
+        break;
+        case StatisticsEventKind::SAMPLE_DATAS:
+        {
+            SampleDatasCountSample sample;
+            EntityId domain;
+            EntityId entity;
+            queue_item_type item = front();
+            sample.src_ts = item.first;
+            try
+            {
+                process_sample_type(domain, entity, EntityKind::DATAWRITER, sample,
+                        item.second->sample_identity_count());
+                database_->insert(domain, entity, sample);
+            }
+            catch (const eprosima::statistics_backend::Exception& e)
+            {
+                //TODO: maybe some logging would be nice
+            }
+        }
+        break;
+        case StatisticsEventKind::PHYSICAL_DATA:
+        {
+            StatisticsPhysicalData item = front().second->physical_data();
+
+            // Take the ID of the Participant from its GUID
+            std::string participant_guid = deserialize_guid(item.participant_guid());
+            auto participants = database_->get_entities_by_guid(EntityKind::PARTICIPANT, participant_guid);
+            if (participants.empty())
+            {
+                std::stringstream msg;
+                msg << "No participant with GUID " << participant_guid << " exists";
+                throw BadParameter(msg.str());
```

Comment:
In general, I'd avoid these temporaries:
```suggestion
                throw BadParameter("No participant with GUID " + participant_guid + " exists");
```